python/tvm/relay/op/contrib/ethosu.py

```diff
@@ -650,12 +650,33 @@ def is_valid(self):
         """
         This function checks whether AvgPool2D has compatible attributes with the NPU
         """
+
+        def check_extra_strides():
```

Comment:
Maybe we can come up with a more self explanatory name for this function? It currently sounds as if some operators have more than one set of strides. Maybe something like `check_same_ifm_and_kernel_shape` or `legalize_large_stride`?